ukm-semantics/test/execution.md

```diff
@@ -27,16 +41,66 @@ endmodule
 module UKM-TEST-EXECUTION
     imports private COMMON-K-CELL
     imports private RUST-EXECUTION-TEST-CONFIGURATION
+    imports private UKM-ENCODING-SYNTAX
     imports private UKM-EXECUTION-SYNTAX
     imports private UKM-HOOKS-BYTES-CONFIGURATION
     imports private UKM-HOOKS-BYTES-SYNTAX
     imports private UKM-HOOKS-HELPERS-SYNTAX
     imports private UKM-HOOKS-STATE-CONFIGURATION
     imports private UKM-HOOKS-UKM-SYNTAX
     imports private UKM-TEST-SYNTAX
+    imports RUST-SHARED-SYNTAX
+    imports private BYTES
 
     syntax Mockable ::= UkmHook
 
+    // The following constructions allows us to build more complex data structures
+    // for mocking tests
+    rule <k> UTH:UKMTestTypeHolder ~> EI:ExecutionItem => EI ~> UTH ... </k>
+    rule <k> UTHL:UKMTestTypeHolderList ~> EI:ExecutionItem => EI ~> UTHL ... </k>
+    rule <k> UTH1:UKMTestTypeHolder ~> UTH2:UKMTestTypeHolder 
+                => (UTH1, UTH2):UKMTestTypeHolderList ... </k> 
+    rule <k> UTH:UKMTestTypeHolder ~> UTHL:UKMTestTypeHolderList 
+                => (UTH, UTHL):UKMTestTypeHolderList ... </k> 
+
+    rule <k> hold_string_from_test_stack => ptr_holder P ... </k>
+         <test-stack> ListItem(P) L:List => L </test-stack>
+    rule <k> ptr_holder ptrValue(_, V) => value_holder V ... </k>
+         
+    rule <k> hold_list_values_from_test_stack => list_ptrs_holder L ~> list_values_holder .List ... </k>
```

Comment:
I would implement this as:
```
rule hold_list_values_from_test_stack => hold_stuff(.List)
rule <k> (.K => V) ~> holdStuff(_) ... </k>
    <test_stack> (ListItem(V) => .List) ... </test_stack>
rule ptrValue(_, V) ~> holdStuff(L) => holdStuff(ListItem(V) L)
```

There is another implementation where you convert the stack to a list of Ptr, then you automatically evaluate that, something like:
```
    rule <k> hold_list_values_from_test_stack => list_values_holder_or_error ptrListToValueList(listToPtrList(Stack), Values) ... </k>
        <test-stack> Stack => .List </test-stack>
        <values> Values </values>
   rule list_values_holder_or_error Values:ValueList => list_values_holder valuesListToList(Values)
```

You would have to either write a `valuesListToList` function or you would need to make `list_values_holder` take a ValueList (honestly, I think that making it take a ValueList is better anyway).

Probably the best solution would be to make some kind of expression list (I'm not sure which one), put it into a constructor, then evaluate the expression list with heating/cooling, but I'm not going to figure out how that should work unless you are actually interested in that. One example would be to create a tuple, i.e., `(ptr(_), ptr(_), ..., ptr(_))`, and have that evaluated automatically (with heating/cooling) to a `tuple(ValueList)`. Or something like that.

Reply:
Added as a TODO. I'll be solving this together with the other issues raised for `ukm-semantics/test/execution.md`.